src/main/java/org/spongepowered/api/service/error/ErrorReport.java

```diff
@@ -0,0 +1,164 @@
+ */
+package org.spongepowered.api.service.error;
+
+import com.google.common.base.Optional;
+import com.google.common.util.concurrent.ListenableFuture;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.util.command.CommandSource;
+
+import java.net.URL;
+
+/**
+ * An error report. This object is mutable.
+ */
+public interface ErrorReport {
+
+    /**
+     * Get the {@link Throwable} that was responsible for creating this error report, if any is present.
+     *
+     * @return The optional throwable for this error report
+     */
+    Optional<Throwable> getCause();
+
+    /**
+     * Check if this error is the result of a misconfiguration on the user's part, or due to an error in programming.
+     * This defaults to true only if the {@link #getCause()} is a {@link UserErrorException}.
+     *
+     * @return Whether or not this is a user error
+     */
+    boolean isUserError();
+
+    /**
+     * Set whether or not this is user error.
+     *
+     * @param isUserError Whether or not this is a user error
+     * @return this
+     */
+    ErrorReport setUserError(boolean isUserError);
+
+    /**
+     * Get a description of the base cause of this error.
+     *
+     * @return The description of this error's base cause
+     */
+    String getCauseDescription();
+
+    /**
+     * Get a section with the given title, creating and appending to the end of the error report as necessary.
+     *
+     * @param title The title to give the section
+     * @return The correct section
+     */
+    Section appendSection(String title);
+
+    /**
+     * Append a {@link Reportable} to this report. If this {@link Reportable} has already been added, it will not be added again.
+     *
+     * @param reportable The object to append information from
+     * @return this
+     */
+    ErrorReport addReportable(Reportable reportable);
+
+    /**
+     * Dispatch this error as a fatal error. Fatal means that the server will be stopped and a full crash report will be generated.
+     *
+     * An exception will be used to pass the result of this error up the call stack. This exception must be passed on.
+     */
+    void dispatchFatal() throws RuntimeException;
+
+    /**
+     * Dispatch this error to a command source. If this command source is not the console, the error report will additionally be mirrored to the
+     * console.
+     *
+     * @param source The source to send information about the error report to
+     */
+    void dispatchToCommandSource(CommandSource source);
+
+    /**
+     * Dispatch this as a non-fatal error that is sent to the console.
+     */
+    void dispatchToConsole();
+
+    /**
+     * Get the text of this error report.
+     *
+     * @return This error report's text
+     */
+    String toText();
+
+    /**
+     * Takes the text content of this error report and posts it to a pastebin service (implementation-determined).
+     * This allows users to more easily send it to the appropriate target.
```

Comment:
should probably be paste upload service or something like that not a name of a specific companies option